docs: update `universal` guide and docs example to v9 (#36483)
angular/angular: aio/content/guide/universal.md

such as JavaScript, image, and style files.

To ensure that clients can only download the files that they are permitted to see, put all client-facing asset files in
the `/dist` folder and only honor requests for files from the `/dist` folder.

The following Node.js Express code routes all remaining requests to `/dist`, and returns a `404 - NOT FOUND` error if the
file isn't found.

<code-example path="universal/server.ts" header="server.ts (static files)" region="static"></code-example>

### Using absolute URLs for HTTP (data) requests on the server

The tutorial's `HeroService` and `HeroSearchService` delegate to the Angular `HttpClient` module to fetch application data.
These services send requests to _relative_ URLs such as `api/heroes`.
In a server-side rendered app, HTTP URLs must be _absolute_ (for example, `https://my-server.com/api/heroes`).
This means that the URLs must be somehow converted to absolute when running on the server and be left relative when running in the browser.

If you are using one of the `@nguniversal/*-engine` packages (such as `@nguniversal/express-engine`), this is taken care for you automatically.

> So, you don't need to do anything to make relative URLs work on the server.

Reviewer: ```suggestion
You don't need to do anything to make relative URLs work on the server.
```

If, for some reason, you are not using an `@nguniversal/*-engine` package, you may need to handle it yourself.

> There are several ways to achieve that.

Reviewer: Unless I am missing something, it appears that we say that there are several ways to handle relative URLs, but only provide one, as it is the recommended way to do it. If that's the case, we can delete line 282.

> The recommended solution, which is also the least intrusive as it does not require any changes to the app, is to pass the full request URL to the `options` argument of [renderModule()](api/platform-server/renderModule) or [renderModuleFactory()](api/platform-server/renderModuleFactory) (depending on what you use to render `AppServerModule` on the server).

Reviewer: ```suggestion
The recommended solution is to pass the full request URL to the `options` argument of [renderModule()](api/platform-server/renderModule) or [renderModuleFactory()](api/platform-server/renderModuleFactory) (depending on what you use to render `AppServerModule` on the server). This option is the least intrusive as it does not require any changes to the app, 
```

Here, "request URL" refers to the URL of the request as a response to which the app is being rendered on the server.
For example, if the client requested `https://my-server.com/dashboard` and you are rendering the app on the server to respond to that request, `options.url` should be set to `https://my-server.com/dashboard`.

> Now, on every HTTP request made as part of rendering the app on the server, Angular will be able to correctly resolve the request URL to an absolute URL, using the provided `options.url`.

Reviewer: ```suggestion
Now, on every HTTP request made as part of rendering the app on the server, Angular can correctly resolve the request URL to an absolute URL, using the provided `options.url`.
```

file isn't found.

<code-example path="universal/server.ts" header="server.ts (static files)" region="static"></code-example>

> ### Using absolute URLs for server requests

Reviewer 2: ```suggestion
### Using absolute URLs for HTTP (data) requests on the server
```

Author: Fixed.
